### ISSUE TITLE
Go 1.23 support with ent schema loader

### ISSUE THREAD

**Issue by GAlexIHU** [ariga/atlas#3068]
- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/ariga/atlas/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

After upgrading to go 1.23 when running `migrate diff` with the ent schema loader it returns a go version error:
```sh
Error: loading schema: entc/load: parse schema dir: /...../internal/ent/schema/balance_snapshot.go:1:1: package requires newer Go version go1.23
```

The actual error originates around here: https://github.com/ent/ent/blob/master/entc/load/load.go#L113 

## Steps to Reproduce 🕹

The environment i'm using is available here https://github.com/openmeterio/openmeter/tree/feat/migrations we're currently integrating atlas into our migration workflow.

To reproduce the error, after cloning just start up a database (our dev stack with `docker compose --profile postgres up -d`) and run `atlas migrate --env local diff somename`. 

## Your Environment 🌎

(We're managing our local env with nix, but here are some versions...)

| Tech        | Version |
| ----------- | ------- |
| Go          | 1.23.0  |
| Atlas         | 0.26.0   |
| Database    | Postgres   |

**Comment by wiliamsouza** [comment]
Same here!
Running `atlas schema inspect -u "ent://ent/schema" --dev-url "sqlite://database.db" -w`

**Comment by Gautam-Diwan** [comment]
Sam here!
Running `sudo atlas migrate diff add_user_balance   --dir "file://ent/migrate/migrations"   --to "ent://ent/schema"   --dev-url "docker://postgres/14/dev?search_path=public"` with no success.
Getting this error: `Error: loading schema: entc/load: parse schema dir: loading package: err: exit status 1: stderr: go: errors parsing go.mod:
$PWD/go.mod:3: invalid go version '1.23.0': must match format 1.23`
No workarounds found as of now

**Comment by a8m** [comment]
Thanks for reporting. It will be handled soon (also in Ent side).

**Comment by a8m** [comment]
Issue was resolved with ent@v0.14.1 and Atlas latest build.

Please reopen if you face any issues with it.
